### ISSUE TITLE
Random weight initialization results in all zero model weights.

### ISSUE THREAD

**Issue by ValMystletainn** [ByteDance-Seed/VeOmni#163]
similar to #118 . I cannot reopen the issue, so I just raise the new issue. If it's not suitable feel free to close it

When train a from scratch, with fsdp2, the model will not init correctly.

However I think this is an unexpected behavior. since I have find the init function call in `veomni/distributed/torch_parallelize.py` function `parallelize_model_fsdp2`

```python
    # Handle meta initialization for FSDP2 (fallback if pre-load not done)
    assert kwargs.get("init_device") == "meta", "Please use init_device: meta for FSDP2"

    if weights_path is None:
        model.to_empty(device="cuda")
        model.init_weights()
```

So I carefully debug to find why there is a call of `model.init_weights()` but still get a from scratch model with weights like just the result of `to_empty` (usually all zero weights).

Actually, is caused by `self.post_init()` in the model `__init__`.

the `self.post_init` of hf `PretrainedModel` will call `self.init_weights()` and when call of `init_weights`, it will set a attribute `_is_hf_initialized` to `True`. 
And when `_is_hf_initialized` is set and be `True`, to `init_weights` method will become short to a dummy return see  [corresponding transformers source code](https://github.com/huggingface/transformers/blob/76fc50a1527a7db593a6057903b749598f7000a9/src/transformers/modeling_utils.py#L2668)

so I think there are two ways to fix it.

1. cancel the `self.post_init()` in all model. but it's not compatible with the existing hf code
2. simple but maybe hacky. turn the fsdp2 meta init into 

```python
    if weights_path is None:
        model.to_empty(device="cuda")
        recursive_set(model, '_is_hf_initialized', False)
        model.init_weights()
```

**Comment by Luosuu** [comment]
Thank you for the insightful report. I will take a look. IMO, maybe we should check `_is_hf_initialized` in `init_weights()`
